### PR TITLE
edbrowse: Fix compilation against newer libcurl

### LINKED ISSUE #53426: edbrowse fails to build
## Issue description

`edbrowse` fails to build with the following error:
```
gcc -DEDBROWSE_ON_LINUX -Wall   -c -o http.o http.c
http.c: In function ‘ebcurl_setError’:
http.c:1558:2: error: duplicate case value
  case CURLE_SSL_CACERT:
  ^~~~
http.c:1557:2: note: previously used here
  case CURLE_PEER_FAILED_VERIFICATION:
  ^~~~
make: *** [<builtin>: http.o] Error 1
```

The two constants have the same value and are the same branch in the switch.

### Steps to reproduce

`nix-build -p edbrowse` fails on my current channel, or you can use `nix-build -E 'let nixpkgs = import (builtins.fetchTarball { url = "https://github.com/NixOS/nixpkgs/archive/487cbfc5634e1f89eb4a72df9ce8e425a0c999f6.tar.gz"; sha256 = "1vw710y1g5b18xjiqhlc2pi3czr1b6zm8cqbigkqz9kxpqmb5awf"; }) {}; in nixpkgs.pkgs.edbrowse'`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.0-13-generic, Ubuntu, 18.10 (Cosmic Cuttlefish)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.3`
 - channels(endgame): `"nixpkgs-19.03pre164964.b0f40b78513"`
 - nixpkgs: `/home/endgame/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by endgame**
###### Motivation for this change

Fixes #53426.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/edbrowse/default.nix [hunk 1]
```diff
@@ -2,7 +2,12 @@
 
 stdenv.mkDerivation rec {
   name = "edbrowse-${version}";
-  version = "3.7.4";
+  # Until edbrowse has a release containing
+  # https://github.com/CMB/edbrowse/commit/5d2b9e21fdf019f461ebe62738d615428d5db963
+  # we won't be able to link against the libcurl in nixpkgs.
+  # Once that happens, revert to using tarballs.
+  gitrev = "c32fdb0c12a16fc031fdfbf8171f8543b3b12e5a";
+  version = "git-${stdenv.lib.substring 0 7 gitrev}";
```
Comment (worldofpeace) [inline pkgs/applications/editors/edbrowse/default.nix]: `version` should be in the format of `unstable-YYYY-MM-DD`
> https://nixos.org/nixpkgs/manual/#sec-package-naming
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23"

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build edbrowse

File: pkgs/applications/editors/edbrowse/default.nix (hunk 1 above)
Comment (endgame) [reply pkgs/applications/editors/edbrowse/default.nix]: Fixed.

File: pkgs/applications/editors/edbrowse/default.nix [hunk 2]
```diff
@@ -2,7 +2,11 @@
 
 stdenv.mkDerivation rec {
   name = "edbrowse-${version}";
-  version = "3.7.4";
+  # Until edbrowse has a release containing
+  # https://github.com/CMB/edbrowse/commit/5d2b9e21fdf019f461ebe62738d615428d5db963
```
[the remark this answers is not in the document]
Comment (endgame) [reply pkgs/applications/editors/edbrowse/default.nix]: I believe so. If that is the preferred way to do this, then I will update the PR.

Would it then be `version = "3.7.4"` again?

[the remark this answers is not in the document]
Comment (worldofpeace) [reply pkgs/applications/editors/edbrowse/default.nix]: > Would it then be `version = "3.7.4"` again?

Yes.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build edbrowse

**Comment by endgame** [comment]
Weird. the sha256 that came from nix-prefetch-url is not the same as the one I see if I or ofborg actually do the build.

**Comment by worldofpeace** [comment]
> Weird. the sha256 that came from nix-prefetch-url is not the same as the one I see if I or ofborg actually do the build.

`fetchpatch` normalizes the patch so that hash isn't the same as what is outputted by `nix-prefetch-url`.

@GrahamcOfBorg build edbrowse

**Comment by worldofpeace** [comment]
Thanks for the fix @endgame

**Pull request merged**